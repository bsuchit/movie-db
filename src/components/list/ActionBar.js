import React, { useState } from "react";
import PropTypes from 'prop-types';
import { Button, Popup, Input } from "semantic-ui-react"; 

import FilterPopup from "components/filter/FilterPopup";
import 'components/components.scss';

const sortByData = [
  {
    name: 'episode_id',
    label: 'Episode'
  },
  {
    name: 'release_date',
    label: 'Release Date'
  }
];

function ActionBar({ onSearch, onSort }) {
  const [searchText, setSearchText] = useState();
  const [sortBySelection, setSortBySelection] = useState();

  const handleOnChange = (e) => {
    e.preventDefault();
    setSearchText(e.target.value);
    onSearch(e.target.value);
  };

  const handleOnFilterSelected = (name, checked) => {
    setSortBySelection(checked ? name : null);
    onSort(name);
};

  return <div>
    <div className="row list_header">
      <Popup 
        content={<FilterPopup
          data={sortByData}
          selection={sortBySelection}
          onSelected={handleOnFilterSelected}
        />} 
        on='click'
        pinned
        trigger={<Button className="sortby_button" content="Sort By" />}
        position="bottom left"
      />
      <Input
        type="text"
        className="search_input"
        placeholder="Type to search..."
        value={searchText}
        onChange={handleOnChange}
        icon='search'
        data-testid='search'
      />
    </div>
  </div>;
}

ActionBar.propTypes = {
  onSort: PropTypes.func,
  onSearch: PropTypes.func
};

export default ActionBar;
