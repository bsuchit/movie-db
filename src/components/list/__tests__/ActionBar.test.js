import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ActionBar from "../ActionBar";

describe('Given ActionBar', () => {
    const mockOnSort = jest.fn();
    const mockOnSearch = jest.fn();

    it('should render correctly with given props', () => {
        const { container } = render(<ActionBar onSearch={mockOnSearch} onSort={mockOnSort} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onSearch when input changed', () => {
        render(<ActionBar onSearch={mockOnSearch} onSort={mockOnSort} />);
        const input = screen.getByTestId('search');
        userEvent.type(input.childNodes[0], 'movie');
        expect(mockOnSearch).toHaveBeenCalledWith('movie');
    });

    it('should call onSort when checkbox checked', async () => {
        render(<ActionBar onSearch={mockOnSearch} onSort={mockOnSort} />);
        const sortButton = screen.getByText('Sort By');
        await userEvent.click(sortButton);
        const checkbox = screen.getByTestId('episode_id');
        userEvent.click(checkbox.childNodes[0]);
        expect(mockOnSort).toHaveBeenCalledWith('episode_id');
    });
});