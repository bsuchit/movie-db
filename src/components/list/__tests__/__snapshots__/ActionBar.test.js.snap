// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given ActionBar should render correctly with given props 1`] = `
<div>
  <div
    class="row list_header"
  >
    <button
      class="ui button sortby_button"
    >
      Sort By
    </button>
    <div
      class="ui icon input search_input"
      data-testid="search"
    >
      <input
        placeholder="Type to search..."
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;
